File: convert-icons.js
#!/usr/bin/env node

/**
 * Convert SVG logo to PNG icons for PWA
 * Requires: npm install sharp
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import sharp from 'sharp';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const publicDir = path.join(__dirname, 'public');
const svgFile = path.join(publicDir, 'icon.svg');

if (!fs.existsSync(svgFile)) {
  console.error(`❌ Error: ${svgFile} not found`);
  process.exit(1);
}

console.log('🎨 Converting your logo to PNG icons...\n');

const sizes = [
  { size: 192, file: 'icon-192.png' },
  { size: 512, file: 'icon-512.png' },
  { size: 192, file: 'icon-maskable-192.png' },
  { size: 512, file: 'icon-maskable-512.png' },
];

(async () => {
  try {
    for (const {size, file} of sizes) {
      const outputPath = path.join(publicDir, file);
      await sharp(svgFile)
        .resize(size, size, {
          fit: 'contain',
          background: { r: 255, g: 255, b: 255, alpha: 1 }
        })
        .png()
        .toFile(outputPath);
      
      console.log(`✅ Created ${file} (${size}x${size})`);
    }
    
    console.log('\n🎉 All icons created successfully!');
    console.log('\nNext steps:');
    console.log('  git add public/icon-*.png');
    console.log('  git commit -m "Add custom GritDocs logo as PWA icons"');
    console.log('  git push');
    
  } catch (error) {
    console.error('❌ Error:', error.message);
    process.exit(1);
  }
})();

File: convert-icons.py
#!/usr/bin/env python3
"""
Convert SVG logo to PNG icons for PWA
"""
import os
import subprocess
from pathlib import Path

# Check if we can use Python's Pillow library
try:
    from PIL import Image
    import io
    HAS_PILLOW = True
except ImportError:
    HAS_PILLOW = False

public_dir = Path('public')
svg_file = public_dir / 'icon.svg'

if not svg_file.exists():
    print(f"❌ Error: {svg_file} not found")
    exit(1)

print(f"📦 Converting {svg_file} to PNG icons...")

# Sizes needed: 192x192 and 512x512 (regular and maskable)
sizes = [
    (192, 'icon-192.png'),
    (512, 'icon-512.png'),
    (192, 'icon-maskable-192.png'),
    (512, 'icon-maskable-512.png'),
]

if HAS_PILLOW:
    print("✅ Using Pillow to convert SVG → PNG")
    from cairosvg import svg2png
    
    try:
        for size, filename in sizes:
            output_path = public_dir / filename
            svg2png(
                url=str(svg_file),
                write_to=str(output_path),
                output_width=size,
                output_height=size
            )
            print(f"  ✓ Created {filename}")
        print("\n✅ All icons created successfully!")
    except Exception as e:
        print(f"⚠️  Pillow method failed: {e}")
        print("Try installing: pip install cairosvg pillow")
else:
    print("⚠️  Python Pillow not installed.")
    print("\n📝 Fallback: Use an online converter")
    print("   1. Go to: https://cloudconvert.com/svg-to-png")
    print("   2. Upload: public/icon.svg")
    print("   3. Download 4 versions:")
    print("      - 192x192 → icon-192.png")
    print("      - 512x512 → icon-512.png")
    print("      - 192x192 (maskable) → icon-maskable-192.png")
    print("      - 512x512 (maskable) → icon-maskable-512.png")
    print("   4. Place all in public/ folder")
